fantasy-docs.md

```diff
@@ -362,6 +393,37 @@ If your function accepts 0 parameters, then it can be visualized easily. But how
 
 This is a really powerful way to let consumers customize the goods you've designed before buying or manufacturing them. For example, you might put a design for a cool 3D printed office chair on thingiverse.com which has a function `main(name: Text)`. This function describes a chair with the given name embossed into the back. When a consumer wants to 3D print it, the 3D printing service will let them input their desired name, view the chair with that name embossed, and then order it.
 
+## Tests
+Functions are marked as tests using the `#[test]` attribute. They're run via the KittyCAD CLI or by the test runner built into the KittyCAD modeling app.
+
+```kcl
+#[test]
+division_by_1_doesnt_change_number() =
+    assert_eq(4/1, 4)
+
+#[test]
+division_by_10() =
+let
+    expected = 10;
+    actual = 100/10;
+in assert_eq(expected, actual)
+```
+The `assert_eq` function will fail the test if the arguments aren't equal. There are similar functions like `assert()` which just checks if its argument is true, or `assert_ne()` which asserts the two are not equal. Tests are run in parallel, because there's no way for two tests to interfere with each other. Test functions cannot take parameters, nor can they return values. If you want to automate tests, consider using smaller functions.
+
+```kcl
+#[test]
+multiplication_by_zero() =
+let
+    n = 100
+    inputs = List.range(0, n) // A list of numbers from `0` to `n`.
```

Comment:
You can reference other variables within a `let...in` expression? That's cool, we might want to highlight that, how `n` can be defined any then used right there.

Reply:
I updated my original let-in example to use this :)

Reply:
Nice I see it now! I think the variable referenced on the right side there just needs to be updated to `can_radius`, currently it says:
```kcl
can_height = can_height * 5
```